lib/vdsm/virt/vmdevices/storage.py

```diff
@@ -1039,6 +1040,10 @@ def _getSourceXML(drive):
     if drive["diskType"] == DISK_TYPE.BLOCK:
         needs_seclabel = True
         source.setAttrs(dev=drive["path"])
+        if 'reservations' in drive and drive["reservations"]:
+            reservations = vmxml.Element('reservations')
+            reservations.set('managed', 'yes')
+            source.appendChild(reservations)
```

Comment:
It would be nicer if we could do:

    if drive.get("reservations"):

but Drive does not implement get(), this code is called with a Drive object or with a dict (e.g. diskReplicate).